halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/EmailDetailsProfileFactoryBuilder.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017 Johan Kasselman.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
+import com.netflix.spinnaker.halyard.deploy.services.v1.ArtifactService;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerArtifact;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerRuntimeSettings;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class EmailDetailsProfileFactoryBuilder {
+  @Autowired
+  protected ArtifactService artifactService;
+
+  @Autowired
+  Yaml yamlParser;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @Setter
+  private String host;
+
+  @Setter
+  private String fromAddress;
+
+  @Setter
+  private SpinnakerArtifact artifact;
+
+  public EmailDetailsProfileFactory build() {
+    return new EmailDetailsProfileFactory(artifact, host, fromAddress);
+  }
+
+  public String getOutputFile(String spinnakerHome) {
+    return Paths.get(spinnakerHome, "path/to/echo.yml").toString();
+  }s
+
+  @EqualsAndHashCode(callSuper = false)
+  @Data
+  public class EmailDetailsProfileFactory extends TemplateBackedProfileFactory {
```

Comment:
This might be a better example: https://github.com/spinnaker/halyard/blob/master/halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/FiatProfileFactory.java#L41

All that's happening is we serialize `authConfig` to yaml, and then append the base contents (from the config bucket) and return that profile. Try doing that with the `notifcations` object instead, and then see what's generated in `~/.hal/default/staging/echo.yml` after running `hal deploy apply`. (That should be fine to run locally, it'll fail once it tries to actually install packages, but after it's generated your config).